### PR TITLE
Add show type in index page, hide used types in create/edit

### PR DESCRIPTION
**Pull request by Glancu**
Add show type in index page, hide used types when create another custom template, remove styleCss as required

### CODE REVIEW

File: src/Provider/EmailCodesProvider.php [hunk 1]
```diff
@@ -10,25 +10,55 @@
...
         $this->emails = $emails;
+        $this->emailTemplateRepository = $emailTemplateRepository;
     }
 
     public function provideWithLabels(): array
+    {
+        return $this->getTypesWithLabels();
+    }
+
+    public function provideWithLabelsNotUsedTypes(): array
```
Comment (senghe) [inline src/Provider/EmailCodesProvider.php:32]: Could you write specs for the method?

File: src/Provider/EmailCodesProvider.php [hunk 2]
```diff
@@ -10,25 +10,55 @@
...
+
+    public function __construct(array $emails, EmailTemplateRepositoryInterface $emailTemplateRepository)
     {
         $this->emails = $emails;
+        $this->emailTemplateRepository = $emailTemplateRepository;
     }
 
     public function provideWithLabels(): array
+    {
+        return $this->getTypesWithLabels();
```
Comment (senghe) [inline src/Provider/EmailCodesProvider.php:29]: Can you remove private method by putting its logic here and use the public one method everywhere the private was used?

File: src/Resources/translations/messages.en.yml [hunk 3]
```diff
@@ -20,3 +20,15 @@ bitbag_sylius_mail_template_plugin:
         not_allowed_twig_property: "Property \"%s\" is not allowed at line %d"
         not_allowed_twig_filter: "Filter \"%s\" is not allowed at line %d"
         not_allowed_twig_tag: "Tag \"%s\" is not allowed at line %d"
+        all_types_have_defined_custom_template: "All types have defined custom template"
+        type:
+            order_confirmation: 'Order Confirmation'
```
Comment (senghe) [inline src/Resources/translations/messages.en.yml:25]: The `'` characters are redundant

File: src/Repository/EmailTemplateRepository.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+final class EmailTemplateRepository implements EmailTemplateRepositoryInterface
+{
+    use RepositoryDecoratorTrait;
+
+    public function __construct(RepositoryInterface $decoratedRepository)
+    {
+        $this->decoratedRepository = $decoratedRepository;
+    }
+
+    public function getAllTypes(): array
```
Comment (senghe) [inline src/Repository/EmailTemplateRepository.php:24]: Can you write integration tests for this method?

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Provider/EmailCodesProvider.php (hunk 1 above)
Comment (Glancu) [reply src/Provider/EmailCodesProvider.php]: Done

File: src/Provider/EmailCodesProvider.php (hunk 2 above)
Comment (Glancu) [reply src/Provider/EmailCodesProvider.php]: Done

File: src/Resources/translations/messages.en.yml (hunk 3 above)
Comment (Glancu) [reply src/Resources/translations/messages.en.yml]: These translations have been removed

File: src/Repository/EmailTemplateRepository.php (hunk 4 above)
Comment (Glancu) [reply src/Repository/EmailTemplateRepository.php:24]: Done

File: tests/Repository/EmailTemplateTest.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    private $emailTemplateRepository;
+
+    protected function setUp(): void
+    {
+        self::bootKernel();
+
+        $this->emailTemplateRepository = self::$container->get('bitbag_sylius_mail_template_plugin.custom_repository.email_template');
+    }
+
+    public function test_return_used_template_email_types(): void
```
Comment (senghe) [inline tests/Repository/EmailTemplateTest.php:27]: Where are fixtures? 🙃

File: tests/Repository/EmailTemplateTest.php [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+declare(strict_types=1);
+
+namespace Tests\BitBag\SyliusMailTemplatePlugin\Repository;
+
+use PHPUnit\Framework\Assert;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+final class EmailTemplateTest extends WebTestCase
+{
+    private $emailTemplateRepository;
```
Comment (senghe) [inline tests/Repository/EmailTemplateTest.php:18]: Where is the property type?

File: src/Form/Type/EmailTemplateType.php [hunk 7]
```diff
@@ -69,4 +79,17 @@ public function getBlockPrefix(): string
     {
         return self::BLOCK_PREFIX;
     }
+
+    private function getAvailableEmailTemplateTypes(EmailTemplateInterface $emailTemplate): array
```
Comment (senghe) [inline src/Form/Type/EmailTemplateType.php:83]: Can you extract this into new service and spec it?

File: src/EventListener/NotFoundTypeCreateEmailTemplateEventListener.php [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        EmailCodesProvider $emailCodesProvider,
+        DataCollectorTranslator $translator,
+        Router $router
+    ) {
+        $this->emailCodesProvider = $emailCodesProvider;
+        $this->translator = $translator;
+        $this->router = $router;
+    }
+
+    public function onKernelRequest(RequestEvent $requestEvent): void
```
Comment (senghe) [inline src/EventListener/NotFoundTypeCreateEmailTemplateEventListener.php:44]: Can you write spec for this class?

**Review by senghe — CHANGES_REQUESTED** [review]

File: tests/Repository/EmailTemplateTest.php (hunk 6 above)
Comment (Glancu) [reply tests/Repository/EmailTemplateTest.php:22]: Fixed

File: src/Form/Type/EmailTemplateType.php (hunk 7 above)
Comment (senghe) [reply src/Form/Type/EmailTemplateType.php]: I don't see specs for it

File: src/EventListener/NotFoundTypeCreateEmailTemplateEventListener.php (hunk 8 above)
Comment (senghe) [reply src/EventListener/NotFoundTypeCreateEmailTemplateEventListener.php:44]: I don't see them

File: tests/Repository/EmailTemplateTest.php (hunk 6 above)
Comment (senghe) [reply tests/Repository/EmailTemplateTest.php:22]: Why you're not using PHP7.4 typing? Does the plugin allow the lower PHP version than 7.4?

File: tests/Repository/EmailTemplateTest.php (hunk 5 above)
Comment (senghe) [reply tests/Repository/EmailTemplateTest.php:39]: Yes. Tests should be independent of application. We can discuss it in a call.

Comment (senghe) [reply tests/Repository/EmailTemplateTest.php:39]: By independend of application I mean, every test has to be able to be ran independently of application state. So in the beginning of integration test you have to create the database state (using fixtures and nelmio alice) and then perform testing code and assertions.

Comment (Glancu) [reply tests/Repository/EmailTemplateTest.php:40]: Done

File: src/Form/Type/EmailTemplateType.php (hunk 7 above)
Comment (Glancu) [reply src/Form/Type/EmailTemplateType.php]: Done

File: src/EventListener/NotFoundTypeCreateEmailTemplateEventListener.php (hunk 8 above)
Comment (Glancu) [reply src/EventListener/NotFoundTypeCreateEmailTemplateEventListener.php:43]: Done

File: tests/Repository/EmailTemplateTest.php [hunk 9]
```diff
@@ -30,10 +40,8 @@ protected function setUp(): void
         $this->emailTemplateRepository = $container->get('bitbag_sylius_mail_template_plugin.custom_repository.email_template');
         $this->emailCodesProvider = $container->get('bitbag_sylius_mail_template_plugin.provider.email_codes');
 
-        $loader = $container->get('fidry_alice_data_fixtures.loader.doctrine');
-        $loader->load([
-            __DIR__ . '/DataFixtures/ORM/email_templates.yml',
-            __DIR__ . '/DataFixtures/ORM/email_templates_translation.yml',
+        $this->loadFixturesFromFiles([
+             'email_templates.yml',
```
Comment (senghe) [inline tests/Repository/EmailTemplateTest.php:44]: Where is the fixtures file placed? I don't see it

**Review by senghe — CHANGES_REQUESTED** [review]

File: tests/Repository/EmailTemplateTest.php (hunk 9 above)
Comment (Glancu) [reply tests/Repository/EmailTemplateTest.php:44]: https://github.com/BitBagCommerce/SyliusMailTemplatePlugin/pull/19/commits/9930e467b8a474c346f21213eb822b9d0efe747f#diff-990cfc9c8a789541c2174347e4357eacb98088200dc829abb1c5204aa8acb0fa

**Review by senghe — APPROVED** [review]

**Pull request merged**